massa-module-cache/src/hd_cache.rs

```diff
@@ -72,6 +72,22 @@ impl HDCache {
         }
     }
 
+    pub fn reset(&mut self) {
```

Comment:
maybe this should return error and avoid unwrap ?
And in execution we can cast it in ExecError

Reply:
Here, the Option should only be None during the reset function (with the `take()`), which can only be called once at a time, so handling the error does not seem needed to me.